rust/src/encodings/plain.rs

```diff
@@ -81,8 +80,15 @@ impl<'a> PlainEncoder<'a> {
     /// Encode primitive values.
     async fn encode_primitive(&mut self, array: &dyn Array) -> Result<usize> {
         let offset = self.writer.tell();
-        let data = array.data().buffers()[0].as_slice();
-        self.writer.write_all(data).await?;
+        let data = array.data();
+        if matches!(array.data_type(), DataType::Boolean) {
+            self.writer.write_all(&data.buffers()[0].as_slice()).await?;
```

Comment:
is there test coverage for this already?

Reply:
added a test.